src/com/owncloud/android/authentication/PatternManager.java

```diff
@@ -0,0 +1,97 @@
+package com.owncloud.android.authentication;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.PowerManager;
+import android.preference.PreferenceManager;
+import android.view.WindowManager;
+
+import com.owncloud.android.BuildConfig;
+import com.owncloud.android.MainApp;
+import com.owncloud.android.ui.activity.PatternLockActivity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by shashvatkedia on 21/12/17.
```

Comment:
Given that this is a new class, please delete this line and add a new comment in the top with the authors and copyright information:

```
/**
 *   ownCloud Android client application
 *
 *   @author YOUR NAME HERE
 *   Copyright (C) 2018 ownCloud GmbH.
 *
 *   This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License version 2,
 *   as published by the Free Software Foundation.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
```